src/stdune/path.ml

```diff
@@ -677,15 +682,17 @@ let relative ?error_loc t fn =
     | In_build_dir p -> in_build_dir (Local.relative p fn ?error_loc)
     | External s -> external_ (External.relative s fn)
 
-let of_string ?error_loc s =
+let parse_string_exn ?error_loc s =
   match s with
   | "" | "." -> in_source_tree Local.root
   | s  ->
     if Filename.is_relative s then
-      make_local_path (Local.of_string s ?error_loc)
+      make_local_path (Local.parse_string_exn s ?error_loc)
     else
       external_ (External.of_string s)
```

Comment:
Should this be calling `External.parse_string_exn`?

Reply:
Indeed.